ESQL: Document BUCKET as a grouping function (#107864)
elastic/elasticsearch: docs/reference/esql/functions/examples/bucket.asciidoc

*Examples*

`BUCKET` can work in two modes: one in which the size of the bucket is computed
based on a buckets count recommendation (four parameters) and a range and
another in which the bucket size is provided directly (two parameters).

> 

Reviewer: Should we also document that `BUCKET` has an alias, `BIN`?

Unlike the earlier example that intentionally filters on a date range, you rarely want to filter on a numeric range.
You have to find the `min` and `max` separately. {esql} doesn't yet have an easy way to do that automatically.

> If the desired bucket size is known in advance, simply provide it as the second
> argument, leaving the range out:

Reviewer: nit: The repetition doesn't read so well.
```suggestion
As for dates, one can omit the range if the desired bucket size is known in advance. Simply provide it as the second
argument:
```

Author: Applied (left out the dates reference, though).

[source.merge.styled,esql]
----
include::{esql-specs}/bucket.csv-spec[tag=docsBucketNumericWithSpan]
----
[%header.monospaced.styled,format=dsv,separator=|]
|===
include::{esql-specs}/bucket.csv-spec[tag=docsBucketNumericWithSpan-result]
|===

> NOTE: When providing the bucket size as the second parameter, its type must be
> of a floating type.

Reviewer: super nit:
```suggestion
NOTE: When providing the bucket size as the second parameter, its type must be
a floating point type.
```
